jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/inputs/CheckboxAssert.java

```diff
@@ -3,35 +3,50 @@
 import com.epam.jdi.light.asserts.generic.UIAssert;
 import com.epam.jdi.light.common.JDIAction;
 import com.epam.jdi.light.material.elements.inputs.Checkbox;
+import com.epam.jdi.tools.Timer;
 import org.hamcrest.Matchers;
 
 import static com.epam.jdi.light.asserts.core.SoftAssert.jdiAssert;
 
 public class CheckboxAssert extends UIAssert<CheckboxAssert, Checkbox> {
 
     @JDIAction("Assert that '{name}' is enabled")
-    @Override
     public CheckboxAssert enabled() {
-        jdiAssert(element().isEnabled(), Matchers.is(true));
+        boolean isEnabled = new Timer(base().getTimeout() * 1000L)
+                .wait(() -> element().isEnabled());
+        jdiAssert(isEnabled, Matchers.is(true));
         return this;
     }
 
     @JDIAction("Assert that '{name}' is disabled")
-    @Override
     public CheckboxAssert disabled() {
-        jdiAssert(element().isDisabled(), Matchers.is(true));
+        boolean isDisabled = new Timer(base().getTimeout() * 1000L)
+                .wait(() -> element().isDisabled());
+        jdiAssert(isDisabled, Matchers.is(true));
         return this;
     }
 
     @JDIAction("Assert that '{name}' is checked")
     public CheckboxAssert checked() {
-        jdiAssert(element().isChecked(), Matchers.is(true));
+        boolean isChecked = new Timer(base().getTimeout() * 1000L)
+                .wait(() -> element().isChecked());
+        jdiAssert(isChecked, Matchers.is(true));
         return this;
     }
 
     @JDIAction("Assert that '{name}' is not checked")
     public CheckboxAssert unChecked() {
-        jdiAssert(element().isNotChecked(), Matchers.is(true));
+        boolean isUnchecked = new Timer(base().getTimeout() * 1000L)
+                .wait(() -> element().isNotChecked());
+        jdiAssert(isUnchecked, Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has '{className}' class")
+    public CheckboxAssert hasClass(String className) {
+        boolean hasClass = new Timer(1000L)
+                .wait(() -> element().hasClass(className));
```

Comment:
Do we really need here wait?

Reply:
We need this wait because sometimes test goes faster than clicking on checkbox and changing classes, which causes failure in assert.